packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -127,7 +272,7 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
       return new AnimationController(
         duration: kThemeAnimationDuration,
         vsync: this,
-      )..addListener(_rebuild);
+      )..addListener(() => setState((){}));
```

Comment:
Never have an empty setState.

It should either set the state, or have a detailed comment explaining what it is that the build function depends on that the setState is flagging as having changed (in this case, the animation controller's value, presumably?).